vault/resource_saml_auth_backend_role.go

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+const (
+	fieldBoundSubjects       = "bound_subjects"
+	fieldBoundSubjectsType   = "bound_subjects_type"
+	fieldBoundAttributes     = "bound_attributes"
+	fieldBoundAttributesType = "bound_attributes_type"
+	fieldGroupsAttribute     = "groups_attribute"
+)
+
+var (
+	samlAuthMountFromPathRegex    = regexp.MustCompile("^auth/(.+)/role/[^/]+$")
+	samlAuthRoleNameFromPathRegex = regexp.MustCompile("^auth/.+/role/([^/]+)$")
+
+	samlRoleAPIFields = []string{
+		fieldBoundSubjects,
+		fieldBoundSubjectsType,
+		fieldBoundAttributes,
+		fieldBoundAttributesType,
+		fieldGroupsAttribute,
+	}
+)
+
+func samlAuthBackendRoleResource() *schema.Resource {
+	fields := map[string]*schema.Schema{
+		consts.FieldPath: {
```

Comment:
Should we provide a default value for this?

Reply:
Ah this shouldn't need a default, but the parameter should be `Required` instead of `Optional` here since it is expected that a SAML Backend resource has been configured prior to setting up the role. Good catch! 🙏🏼